SDWebImage/Core/SDImageCache.m

```diff
@@ -532,11 +532,18 @@ - (nullable NSOperation *)queryCacheOperationForKey:(nullable NSString *)key opt
                 // the image is from in-memory cache, but need image data
                 diskImage = image;
             } else if (diskData) {
-                // decode image data only if in-memory cache missed
-                diskImage = [self diskImageForKey:key data:diskData options:options context:context];
-                if (diskImage && self.config.shouldCacheImagesInMemory) {
-                    NSUInteger cost = diskImage.sd_memoryCost;
-                    [self.memoryCache setObject:diskImage forKey:key cost:cost];
+                BOOL shouldCacheToMomery = YES;
```

Comment:
This condition miss the `if (diskImage && self.config.shouldCacheImagesInMemory)`